Interop/ONNX/src/main/java/org/tribuo/interop/onnx/extractors/BERTFeatureExtractor.java

```diff
@@ -0,0 +1,988 @@
+                            double[] featureValues = extractFeatures(buffer, bertDim);
+                            Example<T> tmp = new ArrayExample<>(outputFactory.getUnknownOutput(),featureNames,featureValues);
+                            tmp.setMetadataValue(TOKEN_METADATA,CLASSIFICATION_TOKEN);
+                            examples.add(tmp);
+                        }
+
+                        // iterate the tokens, creating new examples
+                        for (int i = 0; i < tokens.size(); i++) {
+                            double[] featureValues = extractFeatures(buffer, bertDim);
+                            Example<T> tmp = new ArrayExample<T>(output.get(i),featureNames,featureValues);
+                            tmp.setMetadataValue(TOKEN_METADATA,tokens.get(i));
+                            examples.add(tmp);
+                        }
+
+                        // Add the [SEP] token if necessary
+                        if (!stripSentenceMarkers) {
+                            double[] featureValues = extractFeatures(buffer, bertDim);
+                            Example<T> tmp = new ArrayExample<>(outputFactory.getUnknownOutput(),featureNames,featureValues);
+                            tmp.setMetadataValue(TOKEN_METADATA,SEPARATOR_TOKEN);
+                            examples.add(tmp);
+                        }
+
+                        return new SequenceExample<>(examples);
+                    } else {
+                        throw new IllegalStateException("Expected a float tensor, found " + tensor.getInfo().toString());
+                    }
+                } else {
+                    throw new IllegalStateException("Expected OnnxTensor, found " + value.getClass());
+                }
+            }
+        } catch (OrtException e) {
+            throw new IllegalStateException("ORT failed to execute: ", e);
+        }
+    }
+
+    @Override
+    public void close() throws OrtException {
+        if (!closed) {
+            session.close();
+            env.close();
+            closed = true;
+        }
+    }
+
+    /**
+     * Tokenizes the input using the loaded tokenizer, truncates the
+     * token list if it's longer than {@code maxLength} - 2 (to account
+     * for [CLS] and [SEP] tokens), and then passes the token
+     * list to {@link #extractExample}.
+     * @param output The output object.
+     * @param data The input text.
+     * @return An example containing BERT embedding features and the requested output.
+     */
+    @Override
+    public Example<T> extract(T output, String data) {
+        List<String> tokens = tokenize(data);
+        return extractExample(tokens,output);
+    }
+
+    /**
+     * Tokenizes the input using the loaded tokenizer, truncates the
+     * token list if it's longer than {@code maxLength} - 2 (to account
+     * for [CLS] and [SEP] tokens), and then passes the token
+     * list to {@link #extractExample}.
+     * @param data The input text.
```

Comment:
Maybe document what "tag" is supposed to be? 

Also, maybe add another method `process(String sequence1, String sequence2)` to support sentence pair tasks? (Same with the "extract" method)

Reply:
These methods are constrained by Tribuo's text processing API which doesn't have the notion of sentence pairs. When we add structured inputs then there might be API endpoints for that structure, but we've not finished designing that.

Reply:
Ooops, yeah I was missing the `@param`. Thanks.